src/WebJobs.Extensions/Extensions/Warmup/WarmupConfigProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Extensions.Warmup.Trigger;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Warmup
+{
+    [Extension("Warmup")]
+    public class WarmupConfigProvider : IExtensionConfigProvider
+    {
+        private readonly ILoggerFactory _loggerFactory;
+
+        public WarmupConfigProvider(ILoggerFactory loggerFactory)
+        {
+            _loggerFactory = loggerFactory;
+        }
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var logger = _loggerFactory.CreateLogger(LogCategories.CreateTriggerCategory("Warmup"));
+            logger.LogInformation("Initializing Warmup Extension.");
+
+            context
+                .AddBindingRule<WarmupTriggerAttribute>()
+                .AddConverter<string, JObject>((str) => JObject.Parse(str))
+                .AddOpenConverter<string, OpenType.Poco>(typeof(JsonStringToPocoConverter<>))
```

Comment:
Perhaps we should define a Type WarmupContext as the trigger type. This way we define the contract for the data we'll pass through. For non .NET languages, we'll have the required converters (e.g. WarmupContext -> JObject, or string as needed).

Question is what data should WarmupContext contain?

Reply:
Also, I don't think we need to support OpenType.Poco

Reply:
At the moment, the data sent is a JSON with "hostState" in it.

Maybe the type `WarmupContext` could just have that. And in future if we need to expand it, we can.

Reply:
@mathewc, why not support Poco? It can't hurt to enable it from the extension, no?

This way if users do want to use a POCO, they are free to.